### PR TITLE
Datasetonboarding refactor + swagger docs for search/sort API

### PR DESCRIPTION
**Pull request by cdjreekie**
Some additional tidying up of datasetonboarding + tests
Added swagger docs for search and sort API
Default pagination to 10 if maxResult query param not given to match acceptance criteria

Doesn't include changes to bulk-upload or dataset approval (and MDC integration) - which will be a bigger project

### CODE REVIEW

File: src/resources/dataset/datasetonboarding.service.js [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+		if (!isEmpty(pid)) {
+			dataset = await Data.findOne({ pid: { $eq: pid }, activeflag: constants.datasetStatuses.ACTIVE }).lean();
+
+			if (!isEmpty(datasetID)) {
+				dataset = await Data.findOne({ pid: { $eq: datasetID }, activeflag: constants.datasetStatuses.ARCHIVE }).sort({ createdAt: -1 });
+			}
+		} else if (!isEmpty(datasetID)) {
+			dataset = await Data.findOne({ datasetid: { datasetID } }).lean();
+		}
```
Comment (dannitahdruk) [inline src/resources/dataset/datasetonboarding.service.js:176]: seems like something in this `if ... if ... else if` is not ok.

File: src/resources/dataset/datasetonboarding.service.js [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+			uniqueID = parseInt(Math.random().toString().replace('0.', ''));
+			if ((await Data.find({ id: uniqueID }).length) === 0) uniqueID = '';
+		}
+
+		let newVersion = datasetonboardingUtil.incrementVersion([1, 0, 0], datasetToCopy.datasetVersion);
+
+		datasetToCopy.questionAnswers = JSON.parse(datasetToCopy.questionAnswers);
+
+		if (!datasetToCopy.questionAnswers['properties/documentation/description'] && datasetToCopy.description)
+			datasetToCopy.questionAnswers['properties/documentation/description'] = datasetToCopy.description;
```
Comment (dannitahdruk) [inline src/resources/dataset/datasetonboarding.service.js:279]: for consistency this need to be wrapped  like `if(...) { .... } `

File: src/resources/dataset/datasetonboarding.service.js [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+		data.activeflag = 'draft';
+		data.source = 'HDRUK MDC';
+		data.is5Safes = publisherData[0].uses5Safes;
+		data.questionAnswers = JSON.stringify(datasetToCopy.questionAnswers);
+		data.structuralMetadata = datasetToCopy.structuralMetadata;
+		data.percentageCompleted = datasetToCopy.percentageCompleted;
+		data.timestamps.created = Date.now();
+		data.timestamps.updated = Date.now();
+
+		await data.save();
```
Comment (dannitahdruk) [inline src/resources/dataset/datasetonboarding.service.js:300]: repetitive block - maybe we can optimize this.

File: src/middlewares/datasetonboarding.middleware.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ const validateSearchParameters = (req, res, next) => {
 	const datasetStatuses = Object.values(constants.datasetStatuses);
 
 	let {
-		query: { search = '', datasetIndex = 0, maxResults, sortBy = 'latest', sortDirection = 'desc', status },
+		query: { search = '', datasetIndex = 0, maxResults = 10, sortBy = 'latest', sortDirection = 'desc', status },
```
[the remark this answers is not in the document]
Comment (cdjreekie) [reply src/middlewares/datasetonboarding.middleware.js]: done

File: src/resources/dataset/datasetonboarding.service.js (hunk 1 above)
Comment (cdjreekie) [reply src/resources/dataset/datasetonboarding.service.js]: sorted

File: src/resources/dataset/datasetonboarding.service.js (hunk 2 above)
Comment (cdjreekie) [reply src/resources/dataset/datasetonboarding.service.js]: sorted

File: src/resources/dataset/datasetonboarding.service.js (hunk 3 above)
Comment (cdjreekie) [reply src/resources/dataset/datasetonboarding.service.js]: kept this as is, both blocks are quite different so easier to keep this way for now

**Review by dannitahdruk — APPROVED** [review]

**Pull request merged**